### ISSUE TITLE
Type hints for library

### ISSUE THREAD

**Issue by ASMfreaK** [tomerfiliba/plumbum#334]
Please add type hints for your awesome library, so it can use with mypy checking.

**Comment by henryiii** [comment]
Could you suggest places to add them to expedite the process? I'm familiar with them, and love to see that people are beginning to use them. I'll take PRs (Especially easy for type hints, since they are simply new stub files). I use PyCharm, which also would benefit from type hints.

**Comment by ASMfreaK** [comment]
mypy complains about functions wrapped with ```plumbum.cli.Predicate```:
`error: Invalid type "..."`
Shoul I open another issue for this?

**Comment by henryiii** [comment]
Yes, that needs its own issue, it's caused by Plumbum optionally using type hints as part of the API, and that needs to be made compatible with mypy.

**Comment by mdantonio** [comment]
Hello, this issue is open since a lot of time... can I ask if there are updates on type hints for plumbum? I think that types would be very useful.
Thank you a lot!

**Comment by henryiii** [comment]
Happy to take a PR. Currently we still keep compatibility with Python 2 making it a little harder. I was planning to drop Python 2 Jan 1, 2022 unless @tomerfiliba needs it for some reason. Old plumbum will still work on Python 2 past that point, it just won't get any more updates. Tempted to just drop that much sooner, as the next two features that I don't have time to work on are pulling out at least plumbum.colorlib and plumbum.cli into separate packages, and that's easier in Python 3, and after that adding async support, also easier in Python 3.

**Comment by iacobfred** [comment]
<img width="851" alt="Screenshot 2023-10-11 at 8 32 06 PM" src="https://github.com/tomerfiliba/plumbum/assets/25336795/08b07bfd-a048-4d9c-aebc-e6fc359be77a">

Cool library! Bummer there are no types.
